### DISCUSSION TITLE
"If" expression can support multiple statement within { } block

### DISCUSSION THREAD

**Discussion by wutichai-woody** [redpanda-data/connect#2265] [General]
Hi, 
    I'm using benthos since v3. Can mapping in v4 to support If expression with multiple statement within blacket block?
Ex.

{
     "data_list": [
            ...
            ...
            ...
     ]
}

let var1 = root.data_list
root.result = if this.data.exists('code') {
      root.field1 = "x"
      root.field2 = "y"
      root.field3 = this.data_list or $var1
} else {
      root.field1 = "a"
      root.field2 = "b"
}

// result
when true
{
     "result": {
          "field1": "x",
          "field2": "y",
          "field3":  [
            ...
            ...
            ...
          ]
     }
}

when false 
{
     "result": {
          "field1": "a",
          "field2": "b",
     }
}

**Comment by mihaitodor** [comment]
Hey @wutichai-woody I'm really struggling to understand what you're asking. Would you mind formatting your code properly and giving a clear example that I can run with exact input and output? Otherwise, not sure how to help you except maybe point out [pattern matching](https://www.benthos.dev/docs/guides/bloblang/about#pattern-matching).

**Reply by wutichai-woody** [reply]
Hi @mihaitodor, Thanks for your reply. Sorry for unclear question. I prepare clear question as follows.

INPUT : 
{
  "data": {
     "code": "1234"
  },
  "list_a": [
      ...
      ...
      ...
  ]
  "list_b": [
  ...
  ...
  ...
  ]
}

In mapping:
- mapping : |
      root = {}
      let var1 = root.list_a
      let var2 = root.list_b
      root.result = if this.data.exists('code') {
      root.field1 = "x_1001"
      root.field2 = "y_1001"
      root.field3 = $var1
      } else {
      root.field1 = "x_1111"
      root.field2 = "y_1111"
      root.field3 = $var2
      }

OUTPUT: 
when true
{
    "result": {
          "field1": "x_1001",
          "field2": "y_1001",
          "field3": [
               ... value of "list_a" from INPUT
          ]
    }
}

when false
{
    "result": {
          "field1": "x_1111",
          "field2": "y_1111",
          "field3": [
               ... value of "list_b" from INPUT
          ]
    }
}

**Reply by mihaitodor** [reply]
I don't know how you're writing that mapping, but here's a cleaned-up version which I think outputs what you're looking to get:

```coffee
let var1 = this.list_a
let var2 = this.list_b
root.result = if this.data.exists("code") {
    {
        "field1": "x_1001",
        "field2": "y_1001",
        "field3": $var1 
    }
} else {
    {
        "field1": "x_1111",
        "field2": "y_1111",
        "field3": $var2
    }
}
```

Are you looking to make extra changes to it?

**Reply by wutichai-woody** [reply]
Thanks so much @mihaitodor
